File: tests/color/random.py
import unittest
import goopylib.color.random as gp


class RandomColorMethods(unittest.TestCase):
    def test_random_rgb(self):
        self.assertIsInstance(gp.random_rgb(), gp.ColorRGB)

    def test_random_hex(self):
        self.assertIsInstance(gp.random_hex(), gp.ColorHex)

    def test_random_cmyk(self):
        self.assertIsInstance(gp.random_cmyk(), gp.ColorCMYK)

    def test_random_hsv(self):
        self.assertIsInstance(gp.random_hsv(), gp.ColorHSV)

    def test_random_hsl(self):
        self.assertIsInstance(gp.random_hsl(), gp.ColorHSL)

File: tests/color/random.cpp
#include "color.h"


TEST(ColorRandomTests, ColorFromColorSubclass) {
    gp::Color color = gp::randomColor();
    gp::ColorRGB colorRGB = gp::randomRGB();
    gp::ColorHex colorHex = gp::randomHex();
    gp::ColorCMYK colorCMYK = gp::randomCMYK();
    gp::ColorHSL colorHSL = gp::randomHSL();
    gp::ColorHSV colorHSV = gp::randomHSV();
}
